nexus/src/app/background/tasks/support_bundle_collector.rs

```diff
@@ -83,26 +84,92 @@ fn authz_support_bundle_from_id(id: SupportBundleUuid) -> authz::SupportBundle {
     authz::SupportBundle::new(authz::FLEET, id, LookupType::by_id(id))
 }
 
+// Describes how support bundle data is selected.
+//
+// Multiple values of this enum are joined together into a HashSet.
+// Categories should be additive.
+#[derive(Debug, Clone, Hash, Eq, PartialEq)]
+enum BundleDataCategory {
+    // Collects reconfigurator state (some of the latest blueprints,
+    // information about the target blueprint).
+    Reconfigurator,
+    // Collects info from sled agents, running a handful of
+    // diagnostic commands (e.g., zoneadm, dladm, etc).
+    HostInfo,
+    // Collects sled serial numbers, cubby numbers, and UUIDs.
+    SledCubbyInfo,
+    // Saves task dumps from SPs.
+    SpDumps,
```

Comment:
It's a bit unfortunate that ereports are not a `BundleDataCategory` but are instead represented by the presence of the `EreportFilters` struct. i'm not convinced we should change anything, since the filters struct is more expressive, but...it feels like a bit of a shame.